### DISCUSSION TITLE
[Fixed][License-Eye] skywalking-eyes doesn't recognize absolute path in .gitignore

### DISCUSSION THREAD

**Discussion by spacewander** [apache/skywalking#7542] [Ideas]
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
- [x] Bug
- [ ] Requirement
- [ ] Feature or performance improvement
___
### Bug
- Which version of SkyWalking, OS, and JRE?

- Which company or project?

skywalking-eyes

- What happened?

Take https://github.com/apache/apisix/ as an example.
git clone this repo, touch conf/apisix-aa.yaml.
You will find this file is ignored by `.gitignore`, in absolute path format `/conf/apisix-*.yaml`.

Then you can run `docker run -it --rm -v $(pwd):/github/workspace apache/skywalking-eyes header check`, you will find out this file is not ignored.

**Comment by wu-sheng** [comment]
Why is an git ignored hosted in the repo? What is the point?

**Comment by kezhenxu94** [comment]
> Why is an git ignored hosted in the repo? What is the point?

A case I can think about is to put some files in the codebase that are meant to be modified ever

**Comment by wu-sheng** [comment]
This seems a very not-typical operation. I want to hear what @gxthrj says. Move this discussion.

**Comment by spacewander** [comment]
In some cases, people may need to run a license check locally before pushing the commit. Since skywalking-eyes provides a way to run the license check locally (see the readme), I think it is expected behavior.

Here is a discussion about this requirement: https://github.com/apache/apisix/pull/4872#discussion_r693865175

And it is quite strange that skywalking-eyes ignores relative path in .gitignore (so it is designed to ignore git ignored files?), but doesn't handle absolute path...

**Reply by wu-sheng** [reply]
> so it is designed to ignore git ignored files?

I don't think skywalking-eyes has or should have interaction with git or git ignore config. @kezhenxu94 CMIIW. 

sw-eyes provides `paths-ignore` for your ignorance.

**Reply by spacewander** [reply]
Actually, I found this in the stdout:
```
DEBUG Add ignore path from .gitignore: /conf/apisix-*.yaml
```

skywalking-eyes has add the path, but it doesn't take effect. You can find it via `docker run -it --rm -v $(pwd):/github/workspace apache/skywalking-eyes header check | grep "Add ignore path from .gitignore"`

**Reply by wu-sheng** [reply]
> 7. The paths-ignore are the path list that will be ignored by license-eye. By default, .git and the content in .gitignore will be inflated into the paths-ignore list.

I found this at doc, https://github.com/apache/skywalking-eyes#configurations.

So, according to doc, this is a bug. 

But I would be very surprised about this folder includes actually managed files. @spacewander I think this feature was designed to skip local(never submitted files). 

___
@kezhenxu94 @fgksgf I would treat APISIX's use case as a very specific leak case for header check. It is better to provide a `local-mode`(or any better name) to accept git ignore file.

Once you have a resolution, you could create a bug/feature issue with good explanations about how we should fix this. Or we could continue the discussion here.

**Reply by wu-sheng** [reply]
Or reversely, we should add `strict-mode` for the repo CI process. Because at there, should have no exception unless we have explicit configuration.

Of course, this doesn't affect there is a `git ignore file` analyze bug.

**Reply by wu-sheng** [reply]
https://github.com/apache/skywalking/issues/7543 Created. And PMC members will make the final decision.(Decision doesn't mean they have to fix by themselves)

**Reply by wu-sheng** [reply]
Update, Zhenxu has fixed this.

**Comment by kezhenxu94** [comment]
> In some cases, people may need to run a license check locally before pushing the commit. Since skywalking-eyes provides a way to run the license check locally (see the readme), I think it is expected behavior.
> 
> 
> 
> Here is a discussion about this requirement: https://github.com/apache/apisix/pull/4872#discussion_r693865175
> 
> 
> 
> And it is quite strange that skywalking-eyes ignores relative path in .gitignore (so it is designed to ignore git ignored files?), but doesn't handle absolute path...
> 
> 
> 
> 

I'm not surprised at all that `/conf/apisix-*.yaml` are not ignored because that so-called "absolute path" is in terms of git managed file system, when the path is inflated into skywalking eye, it is "real absolute path" that is relative to the system-wide `/` root because skywalking eye doesn't know anything about git managed files, it just takes advantages of the `.gitignore` so that users don't need to replicate the content of it, so in your specific case, I think changing `/conf/apisix-*.yaml` in `.gitignore` to `./conf/apisix-*.yaml` or `conf/apisix-*.yaml` might do the trick?

**Reply by wu-sheng** [reply]
How about adding a doc to indicate this? And put our commendations.

**Reply by kezhenxu94** [reply]
Well, that might not make sense as `/conf/apisix-*.yaml` and `conf/apisix-*.yaml` are not exactly the same meaning. Need to rethink about this case

**Reply by wu-sheng** [reply]
OK, take your time. There is nothing urgent.
If we really want to work on this, we should take the path of `.gitignore` file as root path, then `/config/apisix-*.yaml` has the correct root.
